client/app/client/components/auth/LoginForm.tsx

```diff
@@ -0,0 +1,131 @@
+import { EyeIcon, EyeOffIcon, LoaderCircleIcon } from "lucide-react"
+import { useState } from "react"
+import { useForm } from "react-hook-form"
+import { toast } from "sonner"
+import { zodResolver } from "@hookform/resolvers/zod"
+import { z } from "zod"
+
+import { Button } from "~/client/components/ui/button"
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage
+} from "~/client/components/ui/form"
+import { Input } from "~/client/components/ui/input"
+import { useCustomMutation } from "~/client/hook/useCustomMutation"
+import { login } from "~/client/services/auth"
+import { signInFormSchema, SignInType } from "~/types/form"
+import { MFAVerificationForm } from "./MFAVerificationForm"
+
+interface LoginProps {
+  onSuccess: () => void
+}
+
+export function LoginForm({ onSuccess }: LoginProps) {
+  const [showPassword, setShowPassword] = useState(false)
+  const [isMFARequired, setIsMFARequired] = useState(false)
+  const [email, setEmail] = useState("")
+
+  const handleShowPassword = () => setShowPassword((prevState) => !prevState)
+
+  const form = useForm<z.infer<typeof signInFormSchema>>({
+    resolver: zodResolver(signInFormSchema),
+    defaultValues: {
+      email: "",
+      password: ""
+    }
+  })
+
+  const { isLoading, mutate } = useCustomMutation(
+    async (values: SignInType) => {
+      return await login(values)
+    },
+    {
+      onSuccess: (data, values) => {
+        if (data?.requires2fa) {
+          setEmail(values.email)
+          setIsMFARequired(true)
+          toast.info("Un code MFA a été envoyé à votre email.")
+        } else {
+          toast.success(data?.customMessage)
+          onSuccess()
+        }
+      },
+      onError: (error) => {
+        toast.error(error.message)
+      }
+    }
+  )
+
+  const handleSubmit = (values: SignInType) => {
+    mutate(values)
+  }
+
+  if (isMFARequired) {
+    return (
+      <MFAVerificationForm
+        key="mfa"
+        email={email}
+        onSuccess={onSuccess}
+        onBack={() => setIsMFARequired(false)}
```

Comment:
```suggestion
      <MFAVerificationForm
        key="mfa"
        email={email}
        onSuccess={onSuccess}
        onBack={setIsMFARequired}
```

And do the set in the component